commit b1dc2b33689668c75e95f873a42d5aea1f4af1db
Author: dtucker@openbsd.org <dtucker@openbsd.org>
Date:   Mon Jul 13 04:57:14 2015 +0000

    upstream commit
    
    Add "PuTTY_Local:" to the clients to which we do not
     offer DH-GEX. This was the string that was used for development versions
     prior to September 2014 and they don't do RFC4419 DH-GEX, but unfortunately
     there are some extant products based on those versions.  bx2424 from Jay
     Rouman, ok markus@ djm@
    
    Upstream-ID: be34d41e18b966832fe09ca243d275b81882e1d5

diff --git a/compat.c b/compat.c
index 0631024..5ea57ed 100644
--- a/compat.c
+++ b/compat.c
@@ -166,7 +166,8 @@ compat_datafellows(const char *version)
 		  "OSU_1.5alpha3*",	SSH_BUG_PASSWORDPAD },
 		{ "*SSH_Version_Mapper*",
 					SSH_BUG_SCANNER },
-		{ "PuTTY-Release-0.5*," /* 0.50-0.57, DH-GEX in >=0.52 */
+		{ "PuTTY_Local:*,"	/* dev versions < Sep 2014 */
+		  "PuTTY-Release-0.5*," /* 0.50-0.57, DH-GEX in >=0.52 */
 		  "PuTTY_Release_0.5*,"	/* 0.58-0.59 */
 		  "PuTTY_Release_0.60*,"
 		  "PuTTY_Release_0.61*,"
